Utilities/macrecovery/macrecovery.py

```diff
@@ -205,27 +199,41 @@ def save_image(url, sess, filename='', directory=''):
     }
 
     if not os.path.exists(directory):
-        os.mkdir(directory)
+        os.makedirs(directory)
 
     if filename == '':
         filename = os.path.basename(purl.path)
-    if filename.find('/') >= 0 or filename == '':
+    if filename.find(os.sep) >= 0 or filename == '':
         raise RuntimeError('Invalid save path ' + filename)
 
-    print(f'Saving {url} to {directory}/{filename}...')
+    print(f'Saving {url} to {directory}{os.sep}{filename}...')
 
     with open(os.path.join(directory, filename), 'wb') as fh:
         response = run_query(url, headers, raw=True)
+        headers = dict(response.headers)
+        totalsize = -1
+        for header in headers:
+            if header.lower() == 'content-length':
+                totalsize = int(headers[header])
+                break
         size = 0
         while True:
             chunk = response.read(2**20)
             if not chunk:
                 break
             fh.write(chunk)
             size += len(chunk)
-            print(f'\r{size / (2**20)} MBs downloaded...', end='')
+            if totalsize + 1:
```

Comment:
`if totalsize + 1:` possibly unecessarily clever, `if totalsize > 0:` probably preferable (easier to read, for same functionality).